libbeat/beat/pipeline.go

```diff
@@ -76,6 +76,10 @@ type ClientConfig struct {
 	// if the normalization step should be skipped set this to true.
 	SkipNormalization bool
 
+	// By default events are decorated with agent metadata.
+	// To skip adding that metadata set this to true.
+	SkipAgentMetadata bool
```

Comment:
Was not even aware we have these options 👍

Reply:
The settings 'Fields', 'Meta', 'EventMetadata', 'DynamicFields' are legacy fields we will remove in the future in favor of processors (I'm in the middle of introducing `add_fields` and `add_tags` processor).

I think I'd prefer to be able to set the agent metadata in the pipeline constructor, and have this optionally be empty in comparison to build more logic/support for the client to change/overwrite global settings. But I know it's not really possible yet to 'influence' the constructor.

By introducing `SkipAgentMetadata` we loose the Beats meta-data always. Is this really intended? I wonder if we want to change but the beats agent meta-data into another namespace (e.g. `collector.agent...`).

Reply:
> By introducing SkipAgentMetadata we loose the Beats meta-data always. Is this really intended?

Yes, for APM we never want libbeat to set these fields, even if apm-server hasn't set them. 
 Another namespace would be fine for the APM case, as we had planned to put them under `observer`.  In that case, we'd want these to be merged (overwrite would be fine even) as `observer` will have other fields too.

Reply:
Perhaps we can later put the Beat info under @metadata so apm-server could still fetch it.

I suggest for now to move forward with this as the only customer of this for now will be apm-server and we can still improve it later. On the Beats side this should not change anything.